golem-component-service-base/src/repo/component.rs

```diff
@@ -142,50 +142,41 @@ impl<DB: Database> DbComponentRepo<DB> {
 
 #[async_trait]
 impl ComponentRepo for DbComponentRepo<sqlx::Sqlite> {
-    async fn upsert(&self, component: &ComponentRecord) -> Result<(), RepoError> {
+    async fn create(&self, component: &ComponentRecord) -> Result<(), RepoError> {
         let mut transaction = self.db_pool.begin().await?;
 
-        let result = sqlx::query(
-            "SELECT count(component_id) as count FROM components WHERE namespace != $1 AND component_id = $2",
-        )
-        .bind(component.namespace.clone())
-        .bind(component.component_id)
-        .fetch_one(&mut *transaction)
-        .await?;
-
-        let count: i64 = result.get("count");
+        let result = sqlx::query("SELECT namespace FROM components WHERE component_id = $1")
+            .bind(component.component_id)
+            .fetch_optional(&mut *transaction)
+            .await?;
 
-        if count > 0 {
-            return Err(RepoError::Internal("Component id not unique".to_string()));
+        if let Some(result) = result {
+            let namespace: String = result.get("namespace");
+            if namespace != component.namespace {
+                return Err(RepoError::Internal("Component id not unique".to_string()));
```

Comment:
This is not an error - you are creating a component version. Component might already exist.

You can (if you want to) compare `name` and `namespace` here.

I would suggest to use `get` here (without namespace - you want to get it from any namespace to compare all the fields).

Reply:
Not sure if I understood the problem here :)

Reply:
Ok, isn't it handled by conflicts with in the sql data table? If not, let's revisit there so that you don't need to explicitly check if a component id exist in some other namespace?

Reply:
Component id is unique already (PK), so it can not exist in some other namespace.

If component id already exists in any namespace - the insert will fail with an error (there is no `ON CONFLICT` any more). So this check is excessive.

Reply:
We should change this check - we should not check the count. We should select the actual component record and validate that the namespace and the name are the same as expected.

Because if we are trying to create the same component_id in the same namespace, but with the different name - it should also be treated as an error.